src/hooks/useLegacyUIURL.ts

```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from "react";
+import { matchPath } from "react-router-dom";
+import get from "lodash/get";
+import { getUiUrl } from "utils/getEnvironmentVariables";
+
+import { routes } from "constants/routes";
+
+export const useLegacyUIURL = (route: string, userId: string) => {
```

Comment:
route and userId do not need to be injected dependencies to the function, unless that would make it easier to test. location and userId are only used for this hook; therefore `useLocation` and `useQuery` can be called within the hook.